portfolio/src/main/webapp/script.js

```diff
@@ -11,18 +11,59 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
+ 
 /**
  * Adds a random greeting to the page.
  */
 function addRandomGreeting() {
   const greetings =
       ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
-
+ 
   // Pick a random greeting.
   const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
+ 
   // Add it to the page.
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+ 
+function getBookInfo() {
+ 
+    const urlHead = "https://www.googleapis.com/books/v1/volumes?q=";
+ 
+    fetch('/book-data').then(response => response.json()).then((bookInfo) => {
+        for (i = 0; i < bookInfo.length; i++) {
+            var query = bookInfo[i].title.replace(/ /g, "+");
+            var url = urlHead + query;
+            searchBooks(bookInfo[i], url);
+        }
+    });
+}
+ 
+function searchBooks(book, url)
+{
+   
+   $.getJSON(url, function (json)
+   {
+        var info = json.items[0].volumeInfo;
+        var title = book.title;
+        var reviews = book.reviews;
+        var author = info.authors;
+        var language = info.language;
+        var categories = info.categories;
+        var description = info.description;
+        var infoLink = info.infoLink;
+        var pageCount = info.pageCount;
+        var publishedDate = info.publishedDate;
+        var publisher = info.publisher;
+        var maturityRating = info.maturityRating;
+ 
```

Comment:
random question: is there a genre label we could use?

Reply:
I believe the "categories" property gives you something like a list of genres from the Books API. And the CSV file gives you a single genre (which I haven't pulled so far) from Goodreads. I can add the genre property to the Book object for now and we can figure out how to use both/either for the recommendations later!

Reply:
ah okay that makes sense! :)